cmd/wio/commands/run/dependencies/target-resolve.go

```diff
@@ -0,0 +1,158 @@
+    name         string
+    isFlag       bool
+    globalsGiven []string
+    otherGiven   []string
+    globals      []string
+    required     []string
+    included     []string
+    onlyGlobal   bool
+    onlyRequired bool
+}
+
+type parentGivenInfo struct {
+    flags          []string
+    definitions    []string
+    linkVisibility string
+    linkFlags      []string
+}
+
+func fillFlagsDefinitions(info flagsDefinitionsInfo) ([]string, error) {
+    var all []string
+
+    global, err := fillGlobal(info.globalsGiven, info.globals, info.name)
+    if err != nil {
+        return nil, err
+    } else {
+        all = utils.AppendIfMissing(all, global)
+    }
+
+    if !info.onlyGlobal && len(info.required) > 0 {
+        required, other, err := fillRequired(info.otherGiven, info.required, info.name)
+        if err != nil {
+            return nil, err
+        } else {
+            all = utils.AppendIfMissing(all, required)
+            all = utils.AppendIfMissing(all, other)
+        }
+    } else {
+        acceptName := "flags"
+        if !info.isFlag {
+            acceptName = "definitions"
+        }
+
+        if info.onlyGlobal && info.onlyRequired {
+            log.Warnln(fmt.Sprintf("%s only accepts global %s and required %s. Accepting only global %s"+
+                "Ignoring...", info.name, acceptName, acceptName, acceptName))
+        } else if info.onlyRequired {
+            log.Warnln(fmt.Sprintf("%s only accepts global %s but required flags are also requested. "+
+                "Ignoring...", info.name, acceptName))
+        }
+
+        all = utils.AppendIfMissing(all, info.otherGiven)
+    }
+
+    all = utils.AppendIfMissing(all, info.included)
+    return all, nil
+}
+
+func resolveTree(i *resolve.Info, currNode *resolve.Node, parentTarget *Target, targetSet *TargetSet,
+    globalFlags, globalDefinitions []string, parentGiven *parentGivenInfo) error {
+    var err error
+
+    pkg, err := i.GetPkg(currNode.Name, currNode.ResolvedVersion.Str())
+    if err != nil {
+        return err
+    } else if pkg == nil {
+        return errors.Stringf("%s dependency does not exist. Check vendor or wio install", currNode.Name)
+    }
+
+    currTarget := &Target{
+        Name:                  pkg.Config.Name(),
+        Version:               pkg.Config.Version(),
+        Path:                  pkg.Path,
+        FromVendor:            pkg.Vendor,
+        HeaderOnly:            pkg.Config.GetMainTag().GetCompileOptions().IsHeaderOnly(),
+        FlagsVisibility:       pkg.Config.MainTag.Flags.Visibility,
+        DefinitionsVisibility: pkg.Config.MainTag.Definitions.Visibility,
+    }
+
+    // resolve flags
+    if currTarget.Flags, err = fillFlagsDefinitions(flagsDefinitionsInfo{
+        name:         currTarget.Name + "__" + currTarget.Version,
+        globalsGiven: globalFlags,
+        otherGiven:   parentGiven.flags,
+        globals:      pkg.Config.MainTag.Flags.GlobalFlags,
```

Comment:
maybe add a `GetFlags` or `GetDefinitions`? would help with abstracting things later

Reply:
For the target or pkg config?

Reply:
This can be done later on based on what wio.yml structure we decide